lib/rops/index.js

```diff
@@ -0,0 +1,113 @@
+const through = require('through2');
+const csv = require('csv-stringify');
+const AWS = require('aws-sdk');
+const archiver = require('archiver');
+
+module.exports = ({ models, s3 }) => {
+
+  const { Establishment, Project, Procedure } = models;
+
+  const S3 = new AWS.S3({
+    apiVersion: '2006-03-01',
+    region: s3.region,
+    accessKeyId: s3.accessKey,
+    secretAccessKey: s3.secret
+  });
+
+  return ({ id, key }) => {
+
+    return new Promise((resolve, reject) => {
+
+      const establishments = csv({ header: true });
+      const projects = csv({ header: true });
+      const procedures = csv({ header: true });
+
+      const meta = {
+        due: 0,
+        submitted: 0
+      };
+
+      Establishment.query()
+        .select(
+          'id',
+          'name',
+          'licenceNumber',
+          'address',
+          'country'
+        )
+        .toKnexQuery()
+        .stream()
+        .pipe(through.obj((record, enc, callback) => callback(null, record)))
+        .pipe(establishments)
+        .on('error', reject);
+
+      Project.query()
+        .select(
+          'rops.id',
+          'projects.licenceNumber',
+          'projects.establishmentId',
+          'projects.title',
+          'projects.status as projectStatus',
+          'projects.issueDate',
+          'projects.expiryDate',
+          'projects.revocationDate',
+          'licenceHolder.firstName',
+          'licenceHolder.lastName',
+          'licenceHolder.email',
+          'licenceHolder.telephone',
+          'rops.*'
```

Comment:
This _might_ want to be reduced down to only the fields that relate to the ROP submission and exclude the fields only used to configure procedure options. Out of scope for now though.